pkg/imagestream/imagestreamhasblob.go

```diff
@@ -37,6 +38,27 @@ func (is *imageStream) HasBlob(ctx context.Context, dgst digest.Digest, requireM
 		return found
 	}
 
+	if layers, err := is.imageStreamGetter.layers(); err == nil {
+		if _, ok := layers.Blobs[dgst.String()]; !ok {
+			return logFound(nil)
+		}
+		copied := make([]imageapiv1.ImageLayer, 0, len(layers.Blobs))
+		for name, blob := range layers.Blobs {
+			copied = append(copied, imageapiv1.ImageLayer{Name: name, LayerSize: blob.LayerSize, MediaType: blob.MediaType})
+		}
+		var firstImage string
+		for name := range layers.Layers {
+			firstImage = name
+			break
+		}
+		return logFound(&imageapiv1.Image{
```

Comment:
Regression. The `requireManaged` was ignored. It will break `config.Pullthrough.Enabled` flag.

https://github.com/openshift/image-registry/pull/101/files#diff-0c454c8033229f303d88c165f82eb0d9L82

Reply:
I’m not sure pullthrough enabled false is something we should support.  In 8 release we’ve never had a reason to trn it off.  At this point, I’d probably prefer to drop the use case.

I don’t really consider the distinction between managed an unmanaged images important anymore.  If a layer is available it should be served.  The image it comes from is irrelevant.

Reply:
Specifically, for the case where we are attempting to find the layers, I think this method should be returning a set of layers that are considered valid (which is added to the cache).

Reply:
OK. Please, open an issue to make pullthrough not optional to address this important change. Because right now this functionality can be turned off and removing this option will need to be written to the log as a warning in the case of `config.Pullthrough.Enabled = false`.

/cc @bparees @dmage